modules/tracking/samples/lasot_benchmark.py

```diff
@@ -0,0 +1,249 @@
+              "MOSSE": (cv.TrackerMOSSE_create(), 1000),
+              "CSRT": (cv.TrackerCSRT_create(), 1000)}
+    return config[tracker_name]
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run LaSOT-based benchmark for visual object trackers")
+    # As a default argument used name of
+    # original dataset folder
+    parser.add_argument("--path_to_dataset", type=str,
+                        default="LaSOTTesting", help="Full path to LaSOT")
+    parser.add_argument("--visualization", action='store_true',
+                        help="Showing process of tracking")
+    args = parser.parse_args()
+
+    # Creating list with names of videos via reading names from txt file
+    video_names = os.path.join(args.path_to_dataset, "testing_set.txt")
+    with open(video_names, 'rt') as f:
+        list_of_videos = f.read().rstrip('\n').split('\n')
+    trackers = [
+        'Boosting', 'MIL', 'KCF', 'MedianFlow', 'GOTURN', 'MOSSE', 'CSRT']
+
+    list_iou = []
+    list_pr = []
+    list_n_pr = []
+
+    # Loop for every tracker
+    for tracker_name in trackers:
+
+        print("Tracker name: ", tracker_name)
+
+        iou_video = np.zeros(21)
+        pr_video = np.zeros(21)
+        n_pr_video = np.zeros(21)
+        iou_thr = np.linspace(0, 1, 21)
+        pr_thr = np.linspace(0, 50, 21)
+        n_pr_thr = np.linspace(0, 0.5, 21)
+
+        # Loop for every video
+        for video_name in list_of_videos:
+
+            tracker, frames_for_reinit = init_tracker(tracker_name)
+            init_once = False
+
+            print("\tVideo name: " + str(video_name))
+
+            # Open specific video and read ground truth for it
+            gt_file = open(os.path.join(args.path_to_dataset, video_name,
+                           "groundtruth.txt"), "r")
+            gt_bb = gt_file.readline().replace("\n", "").split(",")
+            init_bb = gt_bb
+            init_bb = tuple([float(b) for b in init_bb])
+
+            # Creating blob from image sequence
+            video_sequence = sorted(os.listdir(os.path.join(
+                args.path_to_dataset, video_name, "img")))
+
+            # Variables for saving sum of every metric for every frame and
+            # every video respectively
+            iou_list = []
+            pr_list = []
+            n_pr_list = []
+            frame_counter = len(video_sequence)
+
+            # For every frame in video
+            for number_of_the_frame, image in enumerate(video_sequence):
+                frame = cv.imread(os.path.join(
+                    args.path_to_dataset, video_name, "img", image))
+                for i in range(len(gt_bb)):
+                    gt_bb[i] = float(gt_bb[i])
+                gt_bb = tuple(gt_bb)
+
+                # Condition of tracker`s re-initialization
+                if ((number_of_the_frame + 1) % frames_for_reinit == 0) and (
+                        number_of_the_frame != 0):
```

Comment:
Is `number_of_the_frame != 0` necessary?

Reply:
Prevent of using the first frame.

Reply:
I see but seems like `(number_of_the_frame + 1) % frames_for_reinit == 0` is enough for it

Reply:
OK.